storage/innobase/include/ut0counter.h

```diff
@@ -148,7 +136,9 @@ struct MY_ALIGNED(CACHE_LINE_SIZE) ib_counter_t
 	Indexer<Type, N>m_policy;
 
         /** Slot 0 is unused. */
-	Type		m_counter[(N + 1) * (CACHE_LINE_SIZE / sizeof(Type))];
+	std::atomic<Type> m_counter[(N + 1) * (CACHE_LINE_SIZE / sizeof(Type))];
+	static_assert(sizeof(std::atomic<Type>) == sizeof(Type),
+		      "Sizes should match");
```

Comment:
This is something I'm unsure about. I can imagine standard may have reasons not to guarantee size match.

This is just another example of InnoDB overengineering. There should be some way to do it in cleaner and less error prone way. It'd also allow to get rid of `generic_indexer_t`.

So options I could think of (not sure which one works best):
- probably you could fool compiler and align each element via `typedef std::atomic<Type> ib_counter_element_t MY_ALIGNED(CACHE_LINE_SIZE);`
- if not may be try union { std::atomic<Type> m_counter; char pad[CACHE_LINE_SIZE]; }[N]; If unnamed union doesn't work, named should definitely do the trick.
- if you don't like the above, you can add new struct like `struct ib_counter_element_t { std::atomic<Type> m_counter;  } MY_ALIGNED(CACHE_LINE_SIZE);`
- or like `struct ib_counter_element_t { std::atomic<Type> m_counter;  char pad[CACHE_LINE_SIZE - sizeof(std::atomic<Type>)] }`